### PR TITLE
Initial build for data catalog script

### PR DESCRIPTION
**Pull request by meoflaxz**

### CODE REVIEW

File: Data_Catalog/data_catalog.sql [hunk 1]
```diff
@@ -0,0 +1,119 @@
+CREATE OR REPLACE TABLE `x-marketing.data_catalog.data_catalog` AS
```
Comment (izztsfn) [inline Data_Catalog/data_catalog.sql:1]: Use Insert statement instead of Create

File: Data_Catalog/data_catalog.sql [hunk 2]
```diff
@@ -0,0 +1,119 @@
+CREATE OR REPLACE TABLE `x-marketing.data_catalog.data_catalog` AS  
+WITH main_data AS (
+  WITH column_name AS (
```
Comment (izztsfn) [inline Data_Catalog/data_catalog.sql:3]: Avoid using nested CTE

File: Data_Catalog/data_catalog.sql [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  SELECT
+    column_name.*,
+    table_created_time. * EXCEPT (table_schema, table_name)
+  FROM column_name
+  LEFT JOIN table_created_time
+    ON column_name.raw_dataset = table_created_time.table_schema
+    AND column_name.table = table_created_time.table_name
+),
+crm_combined AS (
+  WITH hubspot_source AS (
```
Comment (izztsfn) [inline Data_Catalog/data_catalog.sql:47]: create a separated CTE to combine all of the CRMs to avoid subquerying

**Review by izztsfn — CHANGES_REQUESTED** [review]
Just some formatting changes to make

**Comment by meoflaxz** [comment]
As per latest commit: https://github.com/2x-data-ops/dataops/pull/213/commits/a38efa8af4211e0af3df7be94cd5bc34643d9648

Fixed:
1. Remove both nested CTE
2. Now use INSERT statement instead of CREATE OR REPLACE 

Added:
1. Changed column naming convention
2. Table now has partitioning (table_created) and clustering (_table, _client_name, _crm_source) specs

@izztsfn

**Review by izztsfn — APPROVED** [review]
LGTM. I learned a lot from this script also. Keep up the good work! @meoflaxz

**Pull request merged**